c++/src/vector.cc

```diff
@@ -29,7 +30,7 @@ class Vector {
 
   void push(T val) {
     if (this->length_ >= capacity_) {
-      this->grow();
+      this->reserve(this->capacity_ + 1);
```

Comment:
This threw me off at first because I was expecting the growth factor math, but it's good, as it allows push to say "I don't know whatever you're doing under the hood, I just know I need room for one more that I'm about to push." :+1: